src/vision/oak/camera_oak.py

```diff
@@ -0,0 +1,81 @@
+import depthai as dai
+import open3d as o3d
+import numpy as np
+from vision.oak.imu_tracker import ImuTracker
+from vision.oak.camera_hendler import CameraHendler
+
+
+class CameraOAK:
+    """class for init camera and get data from it."""
+
+    def __init__(self, config: dict, visualize: bool = False):
+        """ Initialize the CameraOAK class.
+        Args:
+            config (dict): The configuration dictionary.
+        """
+        self.handler = CameraHendler(config)
+        self.device = dai.Device(self.handler.pipeline)
+        self.base_time = None
+        self.imu_tracker = ImuTracker()
+        self.visualize = visualize
+        if visualize:
+            self.vis = o3d.visualization.Visualizer()
+            self.vis.create_window()
+            self.line_points = []
+            self.line_set = o3d.geometry.LineSet()
+            self.vis.add_geometry(self.line_set)
+
+    def __del__(self):
+        if self.visualize:
+            self.vis.destroy_window()
+
+    def get_data(self) -> (np.ndarray, o3d.geometry.PointCloud, np.ndarray):
+        """ Get data from the camera.
+        Returns:
+            tuple: (cvColorFrame, pcd, pose) - tuple containing the RGB image, point cloud, and camera position.
+        """
+        imuQueue = self.device.getOutputQueue(name="imu", maxSize=50, blocking=False)
+        pcQueue = self.device.getOutputQueue(name="out", maxSize=4, blocking=False)
+        pose = None
+        pcd = o3d.geometry.PointCloud()
+
+        imuData = imuQueue.get()
```

Comment:
Oak ma też metode `tryGet()` - Try to retrieve message from queue. If no message available, return immediately with nullptr, gdyby się coś wysypało można to zaimplementować.

Reply:
done